src/test/java/unit/Server.java

```diff
@@ -0,0 +1,87 @@
+package unit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.cli.serialization.userfacing.UFStatus;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import harness.TestCommand;
+import harness.baseclasses.ClearContextUnit;
+import java.nio.file.Path;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for the `terra server` commands. */
+@Tag("unit")
+public class Server extends ClearContextUnit {
+  @Test
+  @DisplayName("server status succeeds")
+  void serverStatusSucceeds() throws JsonProcessingException {
+    // `terra server status`
+    String statusMsg =
+        TestCommand.runAndParseCommandExpectSuccess(String.class, "server", "status");
+    assertEquals("OKAY", statusMsg, "server status returns successfully");
+  }
+
+  @Test
+  @DisplayName("status, server list reflect server set")
+  void statusListReflectSet() throws JsonProcessingException {
+    // `terra server set --name=$serverName1`
+    String serverName1 = "terra-verily-dev";
+    TestCommand.runCommandExpectSuccess("server", "set", "--name=" + serverName1);
+
+    // `terra status`
+    UFStatus status = TestCommand.runAndParseCommandExpectSuccess(UFStatus.class, "status");
+    assertEquals(serverName1, status.server.name, "status reflects server set");
+
+    // `terra server list`
+    TestCommand.Result cmd = TestCommand.runCommand("server", "list");
+    assertEquals(0, cmd.exitCode, "server list returned successfully");
+    assertTrue(
+        cmd.stdOut.matches("(?s).*\\*\\s+" + serverName1 + ".*"),
```

Comment:
optional: consider giving an example of what this regex is supposed to match on. ditto other regex